modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/FieldInPageNavigationFormatter.php

```diff
@@ -0,0 +1,201 @@
+   *   Include or not the field.
+   */
+  private function excludeField(string $field_name, array $field, array $fieldgroups) :bool {
+    $type = $this->groupOrField($field_name, $fieldgroups);
+    switch ($type) {
+      case "field":
+        // Exclude empty fields.
+        if ($type === 'field' && !isset($field['#items'])) {
+          return TRUE;
+        }
+
+        break;
+
+      case 'group':
+        // Exclude empty groups.
+        if (count($field) < 1) {
+          return TRUE;
+        }
+        foreach ($field as $value) {
+          // If there is a field with values, don't exclude the group.
+          if (isset($value['#items'])) {
+            return FALSE;
+          }
+        }
+        // Exclude groups with only empty fields.
+        return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Get the label used on the field.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $processed_object
+   *   Entity processed.
+   *
+   * @return string
+   *   Label of the field.
+   */
+  private function fieldGetLabel(string $field_name, array $processed_object): string {
+    $label = '';
+    $type = $this->groupOrField($field_name, $processed_object['#fieldgroups']);
+    switch ($type) {
+      case "field":
+        $label = $this->getFieldLabel($processed_object[$field_name]);
+        break;
+
+      case 'group':
+        $label = $this->getGroupLabel($processed_object['#fieldgroups'][$field_name]);
+        break;
+    }
+    return $label;
+  }
+
+  /**
+   * Returns if the field is a group or a field.
+   *
+   * @param string $field_name
+   *   The name of the field.
+   * @param array $fieldgroups
+   *   List of groups.
+   *
+   * @return string
+   *   The type of field, group or field.
+   */
+  private function groupOrField(string $field_name, array $fieldgroups): string {
+    if (in_array($field_name, array_keys($fieldgroups))) {
+      return 'group';
+    }
+    return 'field';
+  }
+
+  /**
+   * Get the label of the field.
+   *
+   * @param array $field
+   *   Field info.
+   *
+   * @return string
+   *   The label of the field.
+   */
+  private function getFieldLabel(array $field): string {
+    // Show the label if it is not hidden.
+    if (!in_array($field['#label_display'], ['hidden', 'visually_hidden'])) {
+      $title = $field['#title'];
+      if (is_string($title)) {
+        return $field['#title'];
+      }
+      if (is_object($title) && method_exists($title, "__toString")) {
+        return $title->__toString();
+      }
+    }
+    return '';
+  }
+
+  /**
+   * Get the label of the field.
+   *
+   * @param object $group
+   *   Group info.
+   *
+   * @return string
+   *   The label of the group.
+   */
+  private function getGroupLabel($group): string {
+    // If the label is hidden or empty, don't show it.
+    if ($group->label === '' || isset($group->format_settings['show_label']) && $group->format_settings['show_label'] === FALSE) {
```

Comment:
Why don't you just check: empty($group->label)?

Reply:
because of possible false positives like 0